### PR TITLE
[GEOS-8848] Upgrade commons-lang to lang3

### PR DESCRIPTION
**Pull request by bradh**
Upgrade is fairly large, although mostly (not completely) mechanical.

### CODE REVIEW

**Comment by aaime** [comment]
The build failure seem legit:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.3.2:compile (default-compile) on project gs-jms-geoserver: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSServiceHandler.java:[11,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogPostModifyEventHandler.java:[10,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogRemoveEventHandler.java:[12,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSSettingsHandler.java:[9,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogStylesFileHandler.java:[9,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSLoggingHandler.java:[10,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/CatalogUtils.java:[15,31] error: cannot find symbol
[ERROR]  package org.apache.commons.lang3
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSServiceHandler.java:[42,22] error: cannot find symbol
[ERROR]  class JMSServiceHandler
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogPostModifyEventHandler.java:[42,22] error: cannot find symbol
[ERROR]  class JMSCatalogPostModifyEventHandler
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/catalog/JMSCatalogRemoveEventHandler.java:[42,22] error: cannot find symbol
[ERROR]  class JMSCatalogRemoveEventHandler
[ERROR] /home/travis/build/geoserver/geoserver/src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSSettingsHandler.java:[37,22] error: cannot find symbol
[ERROR]  class JMSSettingsHandler...
```

**Comment by jodygarnett** [comment]
Are you in position to rebase @bradh ?

**Comment by aaime** [comment]
Looks like this is in a position to be merged, right?

**Comment by bradh** [comment]
I wasn't sure how much more review was required. I think its fine, but my position is a little biased :-)

File: src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSLoggingHandler.java [hunk 1]
```diff
@@ -37,7 +36,7 @@ protected void omitFields(final XStream xstream) {
     @Override
     public boolean synchronize(LoggingInfo info) throws Exception {
         if (info == null) {
-            throw new NullArgumentException("Incoming object is null");
+            throw new NullPointerException("Incoming object is null");
```
Comment (jodygarnett) [inline src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSLoggingHandler.java]: this appears to be the only substantial difference right? Use of NullPointerException ...

Comment (bradh) [reply src/community/jms-cluster/jms-geoserver/src/main/java/org/geoserver/cluster/impl/handlers/configuration/JMSLoggingHandler.java]: Yes and no. Yes in that lang3 removes NullArgumentException, but no in that there are other changes (e.g. string escapes, signature changes). All the changes are pretty small, just a lot of them.

**Comment by aaime** [comment]
Trusting @jodygarnett approval and merging

**Pull request merged**